tests/pipeline/test_pipeline.py

```diff
@@ -198,6 +198,24 @@ def test_substitute_in_component(self, snapshot: SnapshotTest):
 
         snapshot.assert_match(enriched_pipeline, "test-pipeline")
 
+    @pytest.mark.timeout(10)
```

Comment:
Why do we need this?

Reply:
Update: the infinite loop is avoided by simply counting the iterations and a Warning is displayed every 10 seconds to notify the user what is happening in case the set maximum repetitions are too high

Reply:
this is technically not needed anymore, right? I would be fine with keeping it in as a fallback in case the tested function is changed, so that pytest automatically fails in those cases. timeout could probably further reduced

Reply:
It shouldn't be needed currently, but I would, indeed, prefer to keep it in as a fallback. A silent infinite loop is quite the undesirable 'feature'. 😁

I lowered the time significantly, lmk if you think I should adjust it further